airbyte-to-flow/src/interceptors/airbyte_source_interceptor.rs

```diff
@@ -158,21 +160,34 @@ impl AirbyteSourceInterceptor {
                     cursor_field: stream.default_cursor_field,
                 };
 
-                let source_defined_primary_key = stream.source_defined_primary_key.unwrap_or(Vec::new());
+                let mut source_defined_primary_key = stream.source_defined_primary_key.unwrap_or(Vec::new());
+
+                let recommended_name = stream_to_recommended_name(&stream.name);
+
+                let doc_pk = std::fs::read_to_string(format!("{}/{}{}", STREAM_PATCH_DIR_NAME, recommended_name, STREAM_PK_SUFFIX))
+                    .ok().map(|p| sj::from_str::<sj::Value>(&p)).transpose()?;
+                if let Some(p) = doc_pk {
+                    source_defined_primary_key = p.as_array()
```

Comment:
This made me wonder if we would want to allow a case where we have additive PK patches, and would want to add on to the list of keys from the connector with some new ones. I'm doubtful that it would be worth the effort as it seems we are only having the immediate need to add keys where there weren't any before.

Reply:
@williamhbaker I would imagine that in that case we just manually copy over the full list of primary keys to be honest